torchgeo/trainers/change.py

```diff
@@ -0,0 +1,303 @@
+        return cast(torch.Tensor, self.ce_loss(preds, targets) + self.jaccard_loss(preds, targets))
+
+
+class ChangeDetectionTask(BaseTask):
+    """Change Detection."""
+
+    def __init__(
+        self,
+        model: str = "unet",
+        backbone: str = "resnet50",
+        weights: Optional[Union[WeightsEnum, str, bool]] = None,
+        in_channels: int = 3,
+        num_classes: int = 2,
+        class_weights: Optional[Tensor] = None,
+        labels: Optional[list[str]] = None,
+        loss: str = "ce-jaccard",
+        ignore_index: Optional[int] = None,
+        lr: float = 1e-3,
+        patience: int = 10,
+        freeze_backbone: bool = False,
+        freeze_decoder: bool = False,
+    ) -> None:
+        """Inititalize a new ChangeDetectionTask instance.
+
+        Args:
+            model: Name of the model to use.
+            backbone: Name of the `timm
+                <https://smp.readthedocs.io/en/latest/encoders_timm.html>`__ or `smp
+                <https://smp.readthedocs.io/en/latest/encoders.html>`__ backbone to use.
+            weights: Initial model weights. Either a weight enum, the string
+                representation of a weight enum, True for ImageNet weights, False or
+                None for random weights, or the path to a saved model state dict. FCN
+                model does not support pretrained weights. Pretrained ViT weight enums
+                are not supported yet.
+            in_channels: Number of input channels to model.
+            num_classes: Number of prediction classes.
+            class_weights: Optional rescaling weight given to each
+                class and used with 'ce' loss.
+            labels: Optional labels to use for classes in metrics
+                e.g. ["background", "change"]
+            loss: Name of the loss function, currently supports
+                'ce', 'jaccard', 'focal' or 'focal-jaccard' loss.
+            ignore_index: Optional integer class index to ignore in the loss and
+                metrics.
+            lr: Learning rate for optimizer.
+            patience: Patience for learning rate scheduler.
+            freeze_backbone: Freeze the backbone network to fine-tune the
+                decoder and segmentation head.
+            freeze_decoder: Freeze the decoder network to linear probe
+                the segmentation head.
+
+        .. versionadded: 0.6
+        """
+        if ignore_index is not None and loss == "jaccard":
+            warnings.warn(
+                "ignore_index has no effect on training when loss='jaccard'",
+                UserWarning,
+            )
+
+        self.weights = weights
+        super().__init__()
+
+    def configure_losses(self) -> None:
+        """Initialize the loss criterion.
+
+        Raises:
+            ValueError: If *loss* is invalid.
+        """
+        loss: str = self.hparams["loss"]
+        ignore_index = self.hparams["ignore_index"]
+        if loss == "ce":
```

Comment:
Noting that for change detection BCE may be more appropriate

https://github.com/microsoft/torchgeo/issues/245